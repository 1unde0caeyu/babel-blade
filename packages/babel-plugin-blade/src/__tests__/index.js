import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '../'

const projectRoot = path.join(__dirname, '../../')

expect.addSnapshotSerializer({
  print(val) {
    return val.split(projectRoot).join('<PROJECT_ROOT>/')
  },
  test(val) {
    return typeof val === 'string'
  },
})

const error = code => ({code, error: true})
const noSnapshot = code => ({code, snapshot: false})
const fixture = filename => ({
  fixture: require.resolve(`./fixtures/${filename}`),
})

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {filename: __filename},
  tests: {
    'basic test of functionality': fixture('basic'),
    'injection of fragments': fixture('fragment'),
    // 'does not touch non-blade code': {
    //   snapshot: false,
    //   code: 'const x = notblade`module.exports = "nothing"`;',
    // },
    // 'basic value': 'const x = blade`module.exports = "1"`'
    // 'simple variable assignment':
    //   'blade`module.exports = "var x = \'some directive\'"`',
    // 'object with arrow function': `
    //   const y = blade\`
    //     module.exports = '({booyah: () => "booyah"})'
    //   \`
    // `,
    // 'must export a string': {
    //   code: 'const y = blade`module.exports = {}`',
    //   error: true,
    // },
    // 'blade comment': `
    //   // @blade
    //   const array = ['apple', 'orange', 'pear']
    //   module.exports = array
    //     .map(fruit => \`export const \${fruit} = "\${fruit}";\`)
    //     .join('')
    // `,
    // 'dynamic value that is wrong': {
    //   code: `const x = blade\`module.exports = "\${dynamic}"\``,
    //   error: true,
    // },
    // 'import comment': 'import /* blade */ "./fixtures/assign-one.js"',
    // 'import comment with extra comments after':
    //   'import /* blade */ /* this is extra stuff */ "./fixtures/assign-one.js"',
    // 'import comment with extra comments before':
    //   'import /* this is extra stuff */ /* blade */ "./fixtures/assign-one.js"',
    // 'does not touch import comments that are irrelevant': {
    //   code: 'import /* this is extra stuff */"./fixtures/assign-one.js";',
    //   snapshot: false,
    // },
  },
})

// // This is for any of the exta tests. We give these a name.
// pluginTester({
//   plugin,
//   snapshot: true,
//   babelOptions: {filename: __filename},
//   tests: {
//     'handles some dynamic values': `
//       const three = 3
//       const x = blade\`module.exports = "\${three}"\`
//     `,
//     'accepts babels parser options for generated code': {
//       babelOptions: {
//         filename: __filename,
//         parserOpts: {plugins: ['flow', 'doExpressions']},
//       },
//       code: `
//         // @blade
//         module.exports = "var fNum: number = do { if(true) {100} else {200} };"
//       `,
//     },
//   },
// })
