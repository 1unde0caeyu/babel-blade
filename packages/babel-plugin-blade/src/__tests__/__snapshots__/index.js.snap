// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-blade basic test of functionality: basic test of functionality 1`] = `

import { Connect, query } from 'urql';

const movieQuery = createQuery(); // doesnt take arguments for now
const Movie = ({ id, onClose }) => (
	<div>
		<Connect
			query={query(movieQuery, { id: id })} // razor transpiles into a query string
			children={({ data }) => {
				const DATA = movieQuery(data); // razor(foobar) initializes DATA as a blade, names query as DATA
				const movie = DATA.movie; // \`movie\` is an alias. \`movie\` is also a blade now
				return (
					<div>
						<h2>{movie.gorilla}</h2>
						<p>{movie.monkey}</p>
						<p>{DATA.chimp}</p>
					</div>
				);
			}}
		/>
	</div>
);

      ↓ ↓ ↓ ↓ ↓ ↓

import { Connect, query } from 'urql';

// doesnt take arguments for now
const Movie = ({ id, onClose }) => <div>
		<Connect query={query(\`
query {
  movie {
    gorilla
    monkey
  }
  chimp
}
\`, { id: id })} // razor transpiles into a query string
	children={({ data }) => {
		const DATA = data; // razor(foobar) initializes DATA as a blade, names query as DATA
		const movie = DATA.movie; // \`movie\` is an alias. \`movie\` is also a blade now
		return <div>
						<h2>{movie.gorilla}</h2>
						<p>{movie.monkey}</p>
						<p>{DATA.chimp}</p>
					</div>;
	}} />
	</div>;

`;

exports[`babel-blade injection of fragments: injection of fragments 1`] = `

import { Connect, query } from 'urql';

const fragment = createFragment('Movie');
const Movie = ({ data }) => {
	let result = fragment(data);
	let movie = result.movie;
	return (
		<div>
			<h2>{movie.test.title}</h2>
			<p>{movie.foo}</p>
		</div>
	);
};
Movie.fragment = fragment;

const pageQuery = createQuery(); // create a top-level query
const App = () => (
	// rendering Movie automatically composes \`Movie.fragment\` into the query.
	<Connect
		query={query(pageQuery)}
		children={({ loaded, data }) => {
			let result2 = pageQuery(data);
			return (
				<div>
					<h1>{result2.monkey}</h1>
					<Movie data={result2.movie} />
				</div>
			);
		}}
	/>
);

      ↓ ↓ ↓ ↓ ↓ ↓

import { Connect, query } from 'urql';

const Movie = ({ data }) => {
	let result = data;
	let movie = result.movie;
	return <div>
			<h2>{movie.test.title}</h2>
			<p>{movie.foo}</p>
		</div>;
};
Movie.fragment = \`
fragment movieFragment on Movie {
  movie {
    test {
      title
    }
    foo
  }
}
\`;

// create a top-level query
const App = () =>
// rendering Movie automatically composes \`Movie.fragment\` into the query.
<Connect query={query(\`
query {
  monkey
  movie
}
\`)} children={({ loaded, data }) => {
	let result2 = data;
	return <div>
					<h1>{result2.monkey}</h1>
					<Movie data={result2.movie} />
				</div>;
}} />;

`;
